platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/platformstatus/statuslogic/ReconnectCompleteStatusLogic.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.system.platformstatus.statuslogic;
+
+import com.swirlds.common.system.platformstatus.PlatformStatus;
+import com.swirlds.common.system.platformstatus.PlatformStatusAction;
+import com.swirlds.common.system.platformstatus.PlatformStatusConfig;
+import com.swirlds.common.time.Time;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Class containing the state machine logic for the {@link PlatformStatus#RECONNECT_COMPLETE RECONNECT_COMPLETE} status.
+ */
+public class ReconnectCompleteStatusLogic extends AbstractStatusLogic {
+    /**
+     * Whether a freeze period has been entered
+     */
+    private boolean freezePeriodEntered = false;
+
+    /**
+     * Constructor
+     *
+     * @param time   a source of time
+     * @param config the platform status config
+     */
+    public ReconnectCompleteStatusLogic(@NonNull final Time time, @NonNull final PlatformStatusConfig config) {
+        super(time, config);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @NonNull
+    @Override
+    public PlatformStatus processStatusAction(@NonNull final PlatformStatusAction action) {
+        return switch (action) {
+            case FREEZE_PERIOD_ENTERED -> {
+                freezePeriodEntered = true;
+                yield getStatus();
+            }
+            case FALLEN_BEHIND -> PlatformStatus.BEHIND;
+            case STATE_WRITTEN_TO_DISK -> {
+                // always transition to a new status once a state has been written to disk
+                if (freezePeriodEntered) {
+                    yield PlatformStatus.FREEZING;
+                } else {
+                    yield PlatformStatus.CHECKING;
+                }
+            }
```

Comment:
Something to consider. For reconnect, we need the state that was written to disk to either be the reconnect state, or to be after the reconnect state. Same for freeze. We don't want to transition if the state that was written is for a round that comes before the reconnect/freeze.